scripts/metricsgen/metricsdiff/metricsdiff.go

```diff
@@ -0,0 +1,156 @@
+// metricsdiff is a tool for generating a diff between two different files containing
+// prometheus metrics. metricsdiff outputs which metrics have been added, removed,
+// or have different sets of labels between the two files.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+
+	dto "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+)
+
+func init() {
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, `Usage: %[1]s <path> <path>
+
+Generate the diff between the sets of prometheus metrics peresent in the two files
+in the <path> files. The tool outputs which metrics have been added, removed, or
+use different label sets between the two files.
+
+`, filepath.Base(os.Args[0]))
+		flag.PrintDefaults()
+	}
+}
+
+// MetricsDiff contains the set of metrics that were modified between two files
+// containing prometheus metrics output.
+type MetricsDiff struct {
```

Comment:
I'd consider dropping "Metrics" from this name, it's implicit in the tool name (and if we later move it to a library package, that will probably have "metrics" in the name anyway).
```suggestion
// Diff contains the set of metrics that were modified between two files
// containing prometheus metrics output.
type Diff struct {
```